src/routes.ts

```diff
@@ -6,16 +6,75 @@ import validateRequest from './middlewares/validationResults'
 const router = express.Router()
 
 router.route('/tasks')
+  /**
+   * POST /tasks
+   * @summary Creates a task.
+   * @bodyContent {CreateUpdateTask} application/json
+   * @bodyRequired
+   * @response 201 - Created
+   * @responseContent {Task} 201.application/json
+   * @response 400 - Validation error
+   */
   .post(validateRequest(CreateTaskDto, 'body'),
     task.createTask)
+  /**
+   * GET /tasks
+   * @summary Get all tasks
+   * @queryParam {integer} [skip=0] - The number of items to skip before starting to collect the result
+   * @queryParam {integer} [limit=20] - The numbers of items to return
+   * @queryParam {string} [status] - Filter items by status
+   * @queryParam {string} [importance] - Filter items by importance
+   * @response 200 - The list of the tasks
+   * @responseContent {TasksResponse} 200.application/json
+   */
   .get(validateRequest(GetAllTaskDto, 'query'),
     task.getAllTasks)
 
 router.route('/tasks/:id')
   .all(validateRequest(IdParam, 'params'))
+  /**
+   * GET /tasks/{taskId}
+   * @summary Get task by id
+   * @pathParam {string} taskId - Mongo ID of the task to get.
+   * @response 200 - The list of the tasks
+   * @responseContent {Task} 200.application/json
+   * @response 400 - Invalid task id
+   * @response 404 - Task not found
+   */
   .get(task.getTaskById)
+  /**
+   * PUT /tasks/{taskId}
+   * @summary Replace task by id
+   * @pathParam {string} taskId - Mongo ID of the task to update
```

Comment:
instead of this you could create parameter in yaml and reference it everywhere

```
parameters:
    taskId:
      name: id
      in: path
      description: Task identifier
      required: true
      schema:
        type: string
        format: ObjectId
        
        
  // and usage
  * @paramComponent {taskId}
```